tss-esapi/tests/integration_tests/error_tests/return_code_tests/fapi_tests.rs

```diff
@@ -143,6 +143,60 @@ fn test_invalid_conversions() {
     assert_eq!(
         ReturnCode::try_from(tss_invalid_fapi_rc),
         Err(Error::WrapperError(WrapperErrorKind::InvalidParam)),
-        "Converting invalid FAPI layer resposne code did not produce the expected error"
+        "Converting invalid FAPI layer response code did not produce the expected error"
     );
 }
+
+macro_rules! test_base_error {
+    (BaseError::$base_error:ident) => {
+        let fapi_rc = FapiReturnCode::try_from(BaseError::$base_error).expect(&format!(
+            "Failed to convert {} into FapiReturnCode",
+            std::stringify!(BaseError::$base_error)
+        ));
+
+        assert_eq!(
+            BaseError::$base_error,
+            fapi_rc.base_error(),
+            "`base_error` method did not return the expected value."
+        );
+    };
+}
+
+#[test]
+fn test_base_error_method() {
+    test_base_error!(BaseError::GeneralFailure);
+    test_base_error!(BaseError::NotImplemented);
+    test_base_error!(BaseError::BadReference);
+    test_base_error!(BaseError::BadSequence);
+    test_base_error!(BaseError::IoError);
+    test_base_error!(BaseError::BadValue);
+    test_base_error!(BaseError::NoDecryptParam);
+    test_base_error!(BaseError::NoEncryptParam);
+    test_base_error!(BaseError::Memory);
+    test_base_error!(BaseError::BadContext);
+    test_base_error!(BaseError::NoConfig);
+    test_base_error!(BaseError::BadPath);
+    test_base_error!(BaseError::NotDeletable);
+    test_base_error!(BaseError::PathAlreadyExists);
+    test_base_error!(BaseError::KeyNotFound);
+    test_base_error!(BaseError::SignatureVerificationFailed);
+    test_base_error!(BaseError::HashMismatch);
+    test_base_error!(BaseError::KeyNotDuplicable);
+    test_base_error!(BaseError::PathNotFound);
+    test_base_error!(BaseError::NoCert);
+    test_base_error!(BaseError::NoPcr);
+    test_base_error!(BaseError::PcrNotResettable);
+    test_base_error!(BaseError::BadTemplate);
+    test_base_error!(BaseError::AuthorizationFailed);
+    test_base_error!(BaseError::AuthorizationUnknown);
+    test_base_error!(BaseError::NvNotReadable);
+    test_base_error!(BaseError::NvTooSmall);
+    test_base_error!(BaseError::NvNotWriteable);
+    test_base_error!(BaseError::PolicyUnknown);
+    test_base_error!(BaseError::NvWrongType);
+    test_base_error!(BaseError::NameAlreadyExists);
+    test_base_error!(BaseError::NoTpm);
+    test_base_error!(BaseError::TryAgain);
+    test_base_error!(BaseError::BadKey);
+    test_base_error!(BaseError::NoHandle);
```

Comment:
Similar here, `NotProvisioned` and `AlreadyProvisioned` variants seem to be missing.

Reply:
Fixed